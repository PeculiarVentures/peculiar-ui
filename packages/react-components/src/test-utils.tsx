import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
