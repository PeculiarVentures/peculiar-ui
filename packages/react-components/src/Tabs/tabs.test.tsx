import React from 'react';
import { renderWithWrapper as render } from '../test-utils';
import { Tab, Tabs } from './index';

describe('<Tabs />', () => {
  it('should render with default styles', () => {
    const { asFragment } = render(
      <Tabs value="tab1">
        <Tab id="tab1">Tab-1</Tab>
        <Tab id="tab2">Tab-2</Tab>
      </Tabs>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass className', () => {
    const { asFragment } = render(
      <Tabs value="tab1" className="my-class-name">
        <Tab id="tab1">Tab-1</Tab>
        <Tab id="tab2">Tab-2</Tab>
      </Tabs>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
