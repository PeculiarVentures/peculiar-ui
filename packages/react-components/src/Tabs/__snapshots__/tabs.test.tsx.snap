// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs /> should pass className 1`] = `
<DocumentFragment>
  .emotion-0 {
  white-space: nowrap;
  overflow-x: auto;
}

.emotion-1 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  border: 0;
  border-bottom: 3px solid transparent;
  min-height: var(--pv-size-base-12);
  height: 100%;
  padding: 0 var(--pv-size-base-3);
  background-color: transparent;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-color: var(--pv-color-secondary);
}

.emotion-1:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-7);
}

.emotion-1:not(:disabled) {
  color: var(--pv-color-black);
}

.emotion-1:not(:disabled):hover {
  background-color: var(--pv-color-secondary-tint-5);
}

.emotion-1:not(:disabled):focus-visible {
  background-color: var(--pv-color-secondary-tint-4);
}

.emotion-1:not(:disabled):active {
  background-color: var(--pv-color-secondary-tint-3);
}

.emotion-2 {
  margin: 0;
  color: inherit;
  font-weight: var(--pv-text-s2-weight);
  font-size: var(--pv-text-s2-size);
  line-height: var(--pv-text-s2-height);
  letter-spacing: var(--pv-text-s2-spacing);
}

.emotion-3 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  border: 0;
  border-bottom: 3px solid transparent;
  min-height: var(--pv-size-base-12);
  height: 100%;
  padding: 0 var(--pv-size-base-3);
  background-color: transparent;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-3:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-7);
}

.emotion-3:not(:disabled) {
  color: var(--pv-color-gray-10);
}

.emotion-3:not(:disabled):hover {
  background-color: var(--pv-color-secondary-tint-5);
}

.emotion-3:not(:disabled):focus-visible {
  background-color: var(--pv-color-secondary-tint-4);
}

.emotion-3:not(:disabled):active {
  background-color: var(--pv-color-secondary-tint-3);
}

<div
    class="my-class-name emotion-0"
    role="tablist"
  >
    <button
      aria-selected="true"
      class="emotion-1"
      id="tab1"
      role="tab"
      type="button"
    >
      <span
        class="emotion-2"
      >
        Tab-1
      </span>
    </button>
    <button
      aria-selected="false"
      class="emotion-3"
      id="tab2"
      role="tab"
      type="button"
    >
      <span
        class="emotion-2"
      >
        Tab-2
      </span>
    </button>
  </div>
</DocumentFragment>
`;

exports[`<Tabs /> should render with default styles 1`] = `
<DocumentFragment>
  .emotion-0 {
  white-space: nowrap;
  overflow-x: auto;
}

.emotion-1 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  border: 0;
  border-bottom: 3px solid transparent;
  min-height: var(--pv-size-base-12);
  height: 100%;
  padding: 0 var(--pv-size-base-3);
  background-color: transparent;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-color: var(--pv-color-secondary);
}

.emotion-1:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-7);
}

.emotion-1:not(:disabled) {
  color: var(--pv-color-black);
}

.emotion-1:not(:disabled):hover {
  background-color: var(--pv-color-secondary-tint-5);
}

.emotion-1:not(:disabled):focus-visible {
  background-color: var(--pv-color-secondary-tint-4);
}

.emotion-1:not(:disabled):active {
  background-color: var(--pv-color-secondary-tint-3);
}

.emotion-2 {
  margin: 0;
  color: inherit;
  font-weight: var(--pv-text-s2-weight);
  font-size: var(--pv-text-s2-size);
  line-height: var(--pv-text-s2-height);
  letter-spacing: var(--pv-text-s2-spacing);
}

.emotion-3 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  border: 0;
  border-bottom: 3px solid transparent;
  min-height: var(--pv-size-base-12);
  height: 100%;
  padding: 0 var(--pv-size-base-3);
  background-color: transparent;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-3:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-7);
}

.emotion-3:not(:disabled) {
  color: var(--pv-color-gray-10);
}

.emotion-3:not(:disabled):hover {
  background-color: var(--pv-color-secondary-tint-5);
}

.emotion-3:not(:disabled):focus-visible {
  background-color: var(--pv-color-secondary-tint-4);
}

.emotion-3:not(:disabled):active {
  background-color: var(--pv-color-secondary-tint-3);
}

<div
    class="emotion-0"
    role="tablist"
  >
    <button
      aria-selected="true"
      class="emotion-1"
      id="tab1"
      role="tab"
      type="button"
    >
      <span
        class="emotion-2"
      >
        Tab-1
      </span>
    </button>
    <button
      aria-selected="false"
      class="emotion-3"
      id="tab2"
      role="tab"
      type="button"
    >
      <span
        class="emotion-2"
      >
        Tab-2
      </span>
    </button>
  </div>
</DocumentFragment>
`;
